ci/checks/copyright.py

```diff
@@ -169,39 +197,37 @@ def checkCopyright_main():
     it compares between branches "$PR_TARGET_BRANCH" and "current-pr-branch"
     """
     retVal = 0
-    global ExemptFiles
 
     argparser = argparse.ArgumentParser(
-        "Checks for a consistent copyright header in git's modified files")
-    argparser.add_argument("--update-current-year",
-                           dest='update_current_year',
-                           action="store_true",
-                           required=False,
-                           help="If set, "
-                           "update the current year if a header is already "
-                           "present and well formatted.")
-    argparser.add_argument("--git-modified-only",
-                           dest='git_modified_only',
-                           action="store_true",
-                           required=False,
-                           help="If set, "
-                           "only files seen as modified by git will be "
-                           "processed.")
-
-    (args, dirs) = argparser.parse_known_args()
-    try:
-        ExemptFiles = [re.compile(file) for file in ExemptFiles]
-    except re.error as reException:
-        print("Regular expression error:")
-        print(reException)
-        return 1
+        "Checks for a consistent copyright header in git's modified files"
+    )
+    argparser.add_argument(
+        "--update-current-year",
+        dest="update_current_year",
+        action="store_true",
+        required=False,
+        help="If set, "
+        "update the current year if a header is already "
+        "present and well formatted.",
+    )
+    argparser.add_argument(
+        "--git-modified-only",
+        dest="git_modified_only",
+        action="store_true",
+        required=False,
+        help="If set, "
+        "only files seen as modified by git will be "
+        "processed.",
+    )
+
+    args, dirs = argparser.parse_known_args()
 
     if args.git_modified_only:
```

Comment:
Is this option the only reason that we need GitPython? What happened to the idea of relying on pre-commit entirely to select files?

Reply:
It turns out that we can't use _just_ pre-commit because we need to run this on all files modified between the target branch and the current git staging area. pre-commit can trigger on the files modified in each commit OR all files in the repo (which would modify the end copyright date to `thisYear` for all files in the repo), but has no way to check the history only on files modified since the common ancestor of the target branch.

Reply:
No, I lied. That wasn't the right reason. We can't use _just_ `pre-commit` because we need to be able to run with `--all-files` without updating the year on every file. Typically `pre-commit` only runs on the changes files in a commit, unless you specify `--all-files`. This copyright script should always update "files modified since the last common ancestor with the target branch" regardless of whether you specify `--all-files` or not.

Reply:
Ah right that makes sense. OK yes that makes sense as a limitation then. I suspect that we can do more to simplify the script still, but for this PR I'm happy with keeping the scope of this PR limited.